src/kubernetes.cc

```diff
@@ -164,8 +169,12 @@ json::value KubernetesReader::ComputePodAssociations(const json::Object* pod)
   const json::Object* metadata = pod->Get<json::Object>("metadata");
   const std::string namespace_name = metadata->Get<json::String>("namespace");
 
-  json::value instance_resource =
-      InstanceReader::InstanceResource(environment_).ToJSON();
+  json::value instance_resource;
+  try {
+    instance_resource = store_.LookupResource("").ToJSON();
```

Comment:
Why does this not use instance reader instead?
Where is it documented that the resource ID of the local instance is ""?

Reply:
Since the instance may not exist, I felt that it was counterproductive to expose an instance reader concept anywhere outside of the corresponding poller. The local id of the instance resource being an empty string is a convention (not documented anywhere at the moment — where would it be helpful to document it?).

Reply:
Maybe introduce a constant like kLocalInstanceResourceId?

Reply:
A local constant with a comment sounds good.

Reply:
Good idea. Done.